crates/astria-sequencer/src/checked_actions/bridge/bridge_lock.rs

```diff
@@ -0,0 +1,477 @@
+    action: BridgeLock,
+    tx_signer: TransactionSignerAddressBytes,
+    /// The deposit created from this action.
+    deposit: Deposit,
+}
+
+impl<const PURE_LOCK: bool> CheckedBridgeLockImpl<PURE_LOCK> {
+    #[instrument(skip_all, err(level = Level::DEBUG))]
+    pub(in crate::checked_actions) async fn new<S: StateRead>(
+        action: BridgeLock,
+        tx_signer: [u8; ADDRESS_LEN],
+        tx_id: TransactionId,
+        position_in_tx: u64,
+        state: S,
+    ) -> Result<Self> {
+        state
+            .ensure_base_prefix(&action.to)
+            .await
+            .wrap_err("destination address has an unsupported prefix")?;
+
+        // check that the asset to be transferred matches the bridge account asset.
+        // this also implicitly ensures the recipient is a bridge account.
+        let allowed_asset = state
+            .get_bridge_account_ibc_asset(&action.to)
+            .await
+            .wrap_err(
+                "failed to get bridge account asset ID; destination account is not a bridge \
+                 account",
+            )?;
+        ensure!(
+            allowed_asset == action.asset.to_ibc_prefixed(),
+            "asset ID is not authorized for transfer to bridge account",
+        );
+
+        // Try to construct the `Deposit`.
+        let rollup_id = state
+            .get_bridge_account_rollup_id(&action.to)
+            .await
+            .wrap_err("failed to get bridge account rollup id")?
+            .ok_or_eyre("bridge lock must be sent to a bridge account")?;
+        // Map asset to trace prefixed asset for deposit, if it is not already. The IBC asset cannot
+        // be changed once set in state, so if `map_ibc_to_trace_prefixed_asset` succeeds now it
+        // can't fail later during execution.
+        let deposit_asset = match &action.asset {
+            Denom::TracePrefixed(asset) => asset.clone(),
+            Denom::IbcPrefixed(asset) => state
+                .map_ibc_to_trace_prefixed_asset(asset)
+                .await
+                .wrap_err("failed to map IBC asset to trace prefixed asset")?
+                .ok_or_eyre("mapping from IBC prefixed bridge asset to trace prefixed not found")?,
+        };
+        let deposit = Deposit {
+            bridge_address: action.to,
+            rollup_id,
+            amount: action.amount,
+            asset: deposit_asset.into(),
+            destination_chain_address: action.destination_chain_address.clone(),
+            source_transaction_id: tx_id,
+            source_action_index: position_in_tx,
+        };
+
+        let checked_action = Self {
+            action,
+            tx_signer: tx_signer.into(),
+            deposit,
+        };
+        checked_action.run_mutable_checks(state).await?;
+
+        Ok(checked_action)
+    }
+
+    #[instrument(skip_all, err(level = Level::DEBUG))]
+    pub(in crate::checked_actions) async fn run_mutable_checks<S: StateRead>(
```

Comment:
if `PURE_LOCK` is false, this returns `Ok(())`, so are the mutable checks not necessarily run in `CheckedBridgeLockImpl::new` (line 125)? I'm assuming they always are as the only place this is constructed uses `CheckedBridgeLock::new` (which has PURE_LOCK=true), just want to double check that this is a new invariant for the checked actions